weave-js/src/components/PagePanelComponents/Home/Browse3/components/TraceNavigator/TraceViews/TreeView.tsx

```diff
@@ -0,0 +1,599 @@
+import {Button} from '@wandb/weave/components/Button';
+import {TextField} from '@wandb/weave/components/Form/TextField';
+import {Icon, IconName} from '@wandb/weave/components/Icon';
+import {Tooltip} from '@wandb/weave/components/Tooltip';
+import React, {useEffect, useMemo, useRef, useState} from 'react';
+import {AutoSizer, List} from 'react-virtualized';
+
+import {
+  getCostFromCostData,
+  getTokensFromUsage,
+  TraceStat,
+} from '../../../pages/CallPage/cost';
+import {CallStatusType, StatusChip} from '../../../pages/common/StatusChip';
+import {TraceCallSchema} from '../../../pages/wfReactInterface/traceServerClientTypes';
+import {
+  parseSpanName,
+  traceCallStatusCode,
+} from '../../../pages/wfReactInterface/tsDataModelHooks';
+import TraceScrubber, {ScrubberOption} from '../TraceScrubber';
+import {TraceTreeFlat, TraceViewProps} from './types';
+import {formatDuration, getCallDisplayName} from './utils';
+
+interface FlattenedNode {
+  id: string;
+  call: TraceCallSchema;
+  level: number;
+  isExpanded: boolean;
+  isVisible: boolean;
+  childrenIds: string[];
+  hasDescendantErrors: boolean;
+}
+
+interface TreeNodeProps {
+  node: FlattenedNode;
+  style: React.CSSProperties;
+  focusedCallId?: string;
+  setFocusedCallId: (id: string) => void;
+  rootCallId?: string;
+  setRootCallId: (id: string) => void;
+  onToggleExpand: (id: string) => void;
+  deemphasizeCallIds?: string[];
+  searchQuery?: string;
+  showDuration?: boolean;
+}
+
+type NodeType =
+  | 'agent'
+  | 'tool'
+  | 'llm'
+  | 'model'
+  | 'evaluation'
+  | 'scorer'
+  | 'none';
+
+// Helper function to get call type icon
+const getCallTypeIcon = (type: NodeType): IconName => {
+  switch (type) {
+    case 'agent':
+      return 'robot-service-member';
+    case 'tool':
+      return 'code-alt';
+    case 'llm':
+      return 'forum-chat-bubble';
+    case 'model':
+      return 'model';
+    case 'evaluation':
+      return 'baseline-alt';
+    case 'scorer':
+      return 'number';
+    case 'none':
+      return 'circle';
```

Comment:
I think the circle icon is confusing, it looks to me like an unselected radio button. Maybe we should just omit?

Reply:
👍